cobaya/likelihoods/base_classes/bao.py

```diff
@@ -416,7 +435,11 @@ def rs(self):
         return self.provider.get_param("rdrag") * self.rs_rescale
 
     def logp(self, **params_values):
-        if self.use_grid_2d:
+        if self.use_grid_1d:
+            x = self.theory_fun(self.redshift, self.observable_1)
+            chi2 = float(self.interpolator(x)[0])
```

Comment:
Looks like the problem is wrong extrapolation.
Adding `ext=2` here can make it consistent with CosmoMC_SDSS2020

Reply:
I don't see how that would do it. Doesn't that just raise an error if it's outside the extrapolation range? Did it work for you?

Reply:
Because some likelihoods rise at the edges of parameter space, they extrapolate to extremely large values (very small -loglike), which causes the posterior distribution to reside in these places.
By raising an error can make cobaya to reject these points.
This is a problem I found in my test. I don’t know if it is the same problem you encountered.

Reply:
Because some likelihoods rise at the edges of parameter space, they extrapolate to extremely large values (very small -loglike), which causes the posterior distribution to reside in these places.
By raising an error can make cobaya to reject these points.
This is a problem I found in my test. I don’t know if it is the same problem you encountered.

Reply:
Ok, yes, I can see how that would help in a full MCMC run (where I did see some weirdness), but not for comparing chisquares with CosmoMC.  I'll put in your suggestion, it's a good thing to have.